File: CS/ChartsDemo.Wpf/Utils/MapControlUtils.cs
using DevExpress.Mvvm.UI.Interactivity;
using DevExpress.Xpf.Charts;
using DevExpress.Xpf.Map;
using System;
using System.Globalization;
using System.Linq;
using System.Windows;
using System.Windows.Data;
using System.Windows.Media;

namespace ChartsDemo {
    public class SelectedItemNameBindingBehavior : Behavior<VectorLayer> {
        public static readonly DependencyProperty SelectedItemNameProperty = DependencyProperty.Register("SelectedItemName", typeof(string), typeof(SelectedItemNameBindingBehavior), new PropertyMetadata(null, (d, e) => ((SelectedItemNameBindingBehavior)d).OnSelectedItemNameChanged()));

        public string SelectedItemName {
            get { return (string)GetValue(SelectedItemNameProperty); }
            set { SetValue(SelectedItemNameProperty, value); }
        }

        protected override void OnAttached() {
            base.OnAttached();
            AssociatedObject.Map.SelectionChanged += MapSelectionChanged;
            AssociatedObject.DataLoaded += (_d, _e) => OnSelectedItemNameChanged();
        }

        static string GetName(MapItem mapItem) {
            return (mapItem != null && mapItem.Attributes["NAME"] != null) ? (string)mapItem.Attributes["NAME"].Value : string.Empty;
        }
        void MapSelectionChanged(object sender, MapItemSelectionChangedEventArgs e) {
            SelectedItemName = GetName((MapItem)e.Selection.FirstOrDefault());
        }
        void OnSelectedItemNameChanged() {
            if(AssociatedObject != null && AssociatedObject.Data != null && AssociatedObject.Data.DisplayItems.Any()) 
                AssociatedObject.SelectedItem = AssociatedObject.Data.DisplayItems.FirstOrDefault(x => GetName(x) == SelectedItemName);
        }
    }

    public class ChartPaletteToMapColorsConverter : IValueConverter {
        object IValueConverter.Convert(object value, Type targetType, object parameter, CultureInfo culture) {
            Palette chartColors = (Palette)value;
            DoubleCollection rangeStops = (DoubleCollection)parameter;
            int colorsCount = (int)(rangeStops[rangeStops.Count - 1] - rangeStops[0]) + 1;
            DevExpress.Xpf.Map.ColorCollection mapColors = new DevExpress.Xpf.Map.ColorCollection();
            for(int i = 0; i < colorsCount; i++)
                mapColors.Add(chartColors[i]);
            return mapColors;
        }
        object IValueConverter.ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) {
            return null;
        }
    }
}

File: VB/ChartsDemo.Wpf/Utils/MapControlUtils.vb
Imports DevExpress.Mvvm.UI.Interactivity
Imports DevExpress.Xpf.Charts
Imports DevExpress.Xpf.Map
Imports System
Imports System.Globalization
Imports System.Linq
Imports System.Windows
Imports System.Windows.Data
Imports System.Windows.Media

Namespace ChartsDemo
    Public Class SelectedItemNameBindingBehavior
        Inherits Behavior(Of VectorLayer)

        Public Shared ReadOnly SelectedItemNameProperty As DependencyProperty = DependencyProperty.Register("SelectedItemName", GetType(String), GetType(SelectedItemNameBindingBehavior), New PropertyMetadata(Nothing, Sub(d, e) CType(d, SelectedItemNameBindingBehavior).OnSelectedItemNameChanged()))

        Public Property SelectedItemName() As String
            Get
                Return CStr(GetValue(SelectedItemNameProperty))
            End Get
            Set(ByVal value As String)
                SetValue(SelectedItemNameProperty, value)
            End Set
        End Property

        Protected Overrides Sub OnAttached()
            MyBase.OnAttached()
            AddHandler AssociatedObject.Map.SelectionChanged, AddressOf MapSelectionChanged
            AddHandler AssociatedObject.DataLoaded, Sub(_d, _e) OnSelectedItemNameChanged()
        End Sub

        Private Shared Function GetName(ByVal mapItem As MapItem) As String
            Return If(mapItem IsNot Nothing AndAlso mapItem.Attributes("NAME") IsNot Nothing, CStr(mapItem.Attributes("NAME").Value), String.Empty)
        End Function
        Private Sub MapSelectionChanged(ByVal sender As Object, ByVal e As MapItemSelectionChangedEventArgs)
            SelectedItemName = GetName(CType(e.Selection.FirstOrDefault(), MapItem))
        End Sub
        Private Sub OnSelectedItemNameChanged()
            If AssociatedObject IsNot Nothing AndAlso AssociatedObject.Data IsNot Nothing AndAlso AssociatedObject.Data.DisplayItems.Any() Then
                AssociatedObject.SelectedItem = AssociatedObject.Data.DisplayItems.FirstOrDefault(Function(x) GetName(x) = SelectedItemName)
            End If
        End Sub
    End Class

    Public Class ChartPaletteToMapColorsConverter
        Implements IValueConverter

        Private Function IValueConverter_Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.Convert
            Dim chartColors As Palette = DirectCast(value, Palette)
            Dim rangeStops As DoubleCollection = DirectCast(parameter, DoubleCollection)
            Dim colorsCount As Integer = CInt((rangeStops(rangeStops.Count - 1) - rangeStops(0))) + 1
            Dim mapColors As New DevExpress.Xpf.Map.ColorCollection()
            For i As Integer = 0 To colorsCount - 1
                mapColors.Add(chartColors(i))
            Next i
            Return mapColors
        End Function
        Private Function IValueConverter_ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Return Nothing
        End Function
    End Class
End Namespace
